homework/index.js

```diff
@@ -30,15 +15,115 @@
+    createAndAppend('td', trForks, { text: repo.forks_count });
+    const trUpdate = createAndAppend('tr', tBody);
+    createAndAppend('td', trUpdate, { text: 'Updated:' });
+    createAndAppend('td', trUpdate, {
+      text: new Date(repo.updated_at).toLocaleString(),
+    });
+  }
+
+  function makeContributorsList(contributors) {
+    const rightDiv = document.querySelector('.right-div');
+    createAndAppend('p', rightDiv, { class: 'contributor-header', text: 'Contributions' });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-item',
+        tabindex: '0',
+        'aria-label': contributor.login,
+      });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        class: 'contributor-avatar',
+        height: '48',
+        alt: 'profile photo of contributor',
+      });
+      const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+      createAndAppend('div', contributorData, { text: contributor.login });
+      createAndAppend('div', contributorData, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
+      li.onclick = () => {
+        window.open(contributor.html_url, '_blank');
+      };
+    });
+  }
+
+  async function makeHtmlOfContents(selectedRepo) {
+    document.querySelector('.left-div').innerHTML = '';
+    document.querySelector('.right-div').innerHTML = '';
+    makeTable(selectedRepo);
+    try {
+      const contributorsResponse = await fetch(selectedRepo.contributors_url);
+      const contributorsData = await contributorsResponse.json();
+      makeContributorsList(contributorsData);
+    } catch (err) {
       const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+      createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+    }
+  }
+
+  function mainHtmlConstructor(repos) {
+    makeHeader(repos);
+    const root = document.getElementById('root');
+    const container = createAndAppend('div', root, { id: 'container' });
+    createAndAppend('div', container, { class: 'left-div whiteframe' });
+    createAndAppend('div', container, { class: 'right-div whiteframe' });
+    const selectMenu = document.querySelector('select');
+    makeHtmlOfContents(repos[selectMenu.selectedIndex]);
+    selectMenu.onchange = () => {
+      makeHtmlOfContents(repos[selectMenu.selectedIndex]);
+    };
+  }
+
+  async function main(url) {
+    try {
+      const responseOfHyfRepos = await fetch(url);
+      const repositories = await responseOfHyfRepos.json();
+      if (!responseOfHyfRepos.ok) {
+        throw Error(responseOfHyfRepos.statusText);
```

Comment:
You must always prefix `Error` with `new` (although JavaScript may be forgiving if you forget it):

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error